addon/components/truncate-multiline.js

```diff
@@ -116,16 +116,43 @@ export default Ember.Component.extend(ResizeHandlerMixin, {
   _didTruncate: false,
 
   /**
+   * Resets the component when attributes of a component have changed
+   * @return {Void}
+   */
+  didUpdateAttrs() {
+    this._super(...arguments);
+    this._resetState();
```

Comment:
Should we check if the attribute that was updated is `text` before forcing a rerender? This gets called in other cases where an attribute is changed externally (i.e. external truncation control, hiding/revealing buttons, etc.), and I don't think we should trigger a rerender unless it's necessary.